firedrake/slate/slac/compiler.py

```diff
@@ -88,6 +91,8 @@ def compile_expression(slate_expr, tsfc_parameters=None):
 
     Returns: A `tuple` containing a `SplitKernel(idx, kinfo)`
     """
+    if complex_mode:
+        raise NotImplementedError("SLATE doesn't work in complex mode yet")
```

Comment:
Would it be more appropriate to raise this error higher in the hierarchy, namely in `firedrake/assemble.py` ?